### PR TITLE
feat: Add support for request / response's payload to define HTTP hea…

### PR DESCRIPTION
**Pull request by brasseld**
…ders values

Closes gravitee-io/issues#7333

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `1.10.0-SNAPSHOT.issues-7333-header-payload`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-transformheaders/1.10.0-SNAPSHOT.issues-7333-header-payload/gravitee-policy-transformheaders-1.10.0-SNAPSHOT.issues-7333-header-payload.zip)

### CODE REVIEW

File: src/main/java/io/gravitee/policy/transformheaders/TransformHeadersPolicy.java [hunk 1]
```diff
@@ -76,6 +87,63 @@ public void onResponse(Request request, Response response, ExecutionContext exec
...
+                    context,
+                    (scope == PolicyScope.REQUEST_CONTENT) ? buffer.toString() : null,
+                    (scope == PolicyScope.RESPONSE_CONTENT) ? buffer.toString() : null
+                );
+
+                if (scope == PolicyScope.REQUEST_CONTENT) {
+                    transform(context.request().headers(), context);
+                } else {
+                    transform(context.response().headers(), context);
+                }
```
Comment (ytvnr) [inline src/main/java/io/gravitee/policy/transformheaders/TransformHeadersPolicy.java:130]: It ends in timeout without that (both with postman or debug mode):

```suggestion
                }
                
                if (buffer.length() > 0) {
                    super.write(buffer);
                }

                super.end();
```

**Review by ytvnr — CHANGES_REQUESTED** [review]
It's currently not working for me. (see comment for `BufferedReadWriteStream#end()`)

Can you also do the PRs for `gravitee-api-management/gravitee-apim-distribution/pom.xml `and `release.json` please?

File: src/main/java/io/gravitee/policy/transformheaders/TransformHeadersPolicy.java (hunk 1 above)
Comment (jhaeyaert) [inline src/main/java/io/gravitee/policy/transformheaders/TransformHeadersPolicy.java:130]: I think a call to `super.end()` is missing here no ? Without it I think it breaks the execution chain. WDYT ?

**Comment by gcusnieux** [comment]
Hi @brasseld, 
After the last fix on gateway, I wonder if we should not make an `add` rather than a `set` here: https://github.com/gravitee-io/gravitee-policy-transformheaders/blob/master/src/main/java/io/gravitee/policy/transformheaders/TransformHeadersPolicy.java#L91

**Review by ytvnr — CHANGES_REQUESTED** [review]
is failing because of a wrong dependency in pom.xml

**Review by ytvnr — APPROVED** [review]
lgtm

**Review by phiz71 — APPROVED** [review]

**Pull request merged**